File: p3206/rust/src/main.rs
struct Solution;


impl Solution {
    pub fn number_of_alternating_groups(colors: Vec<i32>) -> i32 {
        let len =colors.len(); 
        if  len<= 3  {
            return 0;
        }
        let mut left_idx = 0;
        let mut middle_idx = 1;
        let mut rigth_idx = 2;
        let mut resulr = 0;
        let mut evaluation = 0;
        while evaluation < len  {
            if (colors[left_idx % len ] == colors[rigth_idx % len])
              && colors[middle_idx % len] != colors[left_idx % len]
              && colors[middle_idx % len] != colors[rigth_idx % len]{
                resulr +=1;    
            }
            rigth_idx +=1;
            middle_idx +=1;
            left_idx +=1;
            evaluation +=1;
        }
        print!("{}",evaluation);
        resulr
    }
}

fn main(){
    
}

File: p3206/cpp/main.cpp
#include <vector>
class Solution {
public:
    int numberOfAlternatingGroups(std::vector<int>& colors) {
        int len = colors.size();
        int leff_idx = 0;
        int middle_idx = 1;
        int right_idx = 2;
        int iter = 0;
        int result = 0;
        while (iter < len )
        {
            if (colors[right_idx % len] == colors[leff_idx % len]
            && colors[middle_idx % len] != colors[leff_idx% len]
            && colors[middle_idx % len] != colors[right_idx % len])
            {
                result ++;
            }
            
            leff_idx++;
            right_idx++;
            middle_idx++;
            iter++;
        }
        
        return result;
    }
};